examples/shapes_and_collections/fancybox_demo.py

```diff
@@ -27,169 +27,77 @@
-                             boxstyle="round,pad=0.1",
-                             mutation_scale=2.,
-                             fc=(1., .8, 1.),
-                             ec=(1., 0.5, 1.))
-
-    ax.add_patch(p_fancy)
-
-    ax.text(0.1, 0.8,
-            ' boxstyle="round,pad=0.1"\n mutation_scale=2',
-            size=10, transform=ax.transAxes)
-
-    # draws control points for the fancy box.
-    # l = p_fancy.get_path().vertices
-    # ax.plot(l[:,0], l[:,1], ".")
-
-    draw_bbox(ax, bb)
-
-
-def test4(ax):
-
-    # When the aspect ratio of the axes is not 1, the fancy box may
-    # not be what you expected (green)
-
-    p_fancy = FancyBboxPatch((bb.xmin, bb.ymin),
-                             abs(bb.width), abs(bb.height),
-                             boxstyle="round,pad=0.2",
-                             fc="none",
-                             ec=(0., .5, 0.), zorder=4)
-
-    ax.add_patch(p_fancy)
-
+    fancy = add_fancy_patch_around(ax, bb, boxstyle="round,pad=0.1")
+    # The boxstyle and its argument can be later modified with set_boxstyle().
+    # Note that the old attributes are simply forgotten even if the boxstyle
+    # name is same.
+    fancy.set_boxstyle("round,pad=0.1,rounding_size=0.2")
+    # or: fancy.set_boxstyle("round", pad=0.1, rounding_size=0.2)
+    ax.set(xlim=(0, 1), ylim=(0, 1), aspect=1,
+           title='boxstyle="round,pad=0.1,rounding_size=0.2"')
+
+    ax = axs[2]
+    # mutation_scale determine overall scale of the mutation, i.e. both pad and
+    # rounding_size is scaled according to this value.
+    fancy = add_fancy_patch_around(
+        ax, bb, boxstyle="round,pad=0.1", mutation_scale=2)
+    ax.set(xlim=(0, 1), ylim=(0, 1), aspect=1,
+           title='boxstyle="round,pad=0.1"\n mutation_scale=2')
+
+    ax = axs[3]
+    # When the aspect ratio of the axes is not 1, the fancy box may not be what
+    # you expected (green).
+    fancy = add_fancy_patch_around(ax, bb, boxstyle="round,pad=0.2")
+    fancy.set(facecolor="none", edgecolor="green")
     # You can compensate this by setting the mutation_aspect (pink).
-    p_fancy = FancyBboxPatch((bb.xmin, bb.ymin),
-                             abs(bb.width), abs(bb.height),
-                             boxstyle="round,pad=0.3",
-                             mutation_aspect=.5,
-                             fc=(1., 0.8, 1.),
-                             ec=(1., 0.5, 1.))
-
-    ax.add_patch(p_fancy)
+    fancy = add_fancy_patch_around(
+        ax, bb, boxstyle="round,pad=0.3", mutation_aspect=0.5)
+    ax.set(xlim=(-.5, 1.5), ylim=(0, 1), aspect=2,
+           title='boxstyle="round,pad=0.3"\nmutation_aspect=.5')
 
-    ax.text(0.1, 0.8,
-            ' boxstyle="round,pad=0.3"\n mutation_aspect=.5',
-            size=10, transform=ax.transAxes)
+    for ax in axs:
+        # Draw the control points of the fancy boxes.
+        for patch in ax.patches:
+            patch.axes.plot(*patch.get_path().vertices.T, ".",
```

Comment:
Maybe extract as helper function at the top so that it distracts less from the actual box drawing.